File: Aula14/ex03.cs
using System;

enum Estacao {
  Primavera, Verão, Outono, Inverno
}

[Flags]
public enum Dias {
  Nenhum = 0, 
  Segunda = 1,
  Terça = 2, 
  Quarta = 4,
  Quinta = 8, 
  Sexta = 16,
  Sábado = 32, 
  Domingo = 64
}

class MainClass {
  public static void Main (string[] args) {
    Estacao x = Estacao.Outono;
    Console.WriteLine(x); 
    Estacao y = (Estacao) 3;
    Console.WriteLine(y); 
    Estacao z = Estacao.Verão | Estacao.Outono;
    Console.WriteLine(z); 

    Dias poo = Dias.Quarta | Dias.Quinta;
    Console.WriteLine(poo);

    if ((Dias.Quarta & poo) != 0)
      Console.WriteLine("Tem aula na quarta");
  }
}

File: Aula14/ex03.java
public class ex03 {
  public static void main(String[] args) {
    Estacao x = Estacao.Outono;
    System.out.println(x);
    if (x == Estacao.Outono)
      System.out.println("Teste");
    // Pesquisar EnumSet
    Dias poo = Dias.Quarta;
    System.out.println(poo);
  }
}

public enum Estacao{
  Primavera(1), Verao(2), Outono(3), Inverno(4);
  public final int valor;
  Estacao(int id){
    this.valor = id;
   }
  //public int getValor(){
  //  return valor;
  //}
}
public enum Dias {
  Nenhum(0), Segunda(1), Terca(2), Quarta(4), Quinta(8), Sexta(16), Sabado(32), Domingo(64);
  public final int valor;
  Dias(int id){
    this.valor = id;
  }
}

